extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/sql/resources/SqlStatementResource.java

```diff
@@ -284,28 +285,25 @@ public Response doGetResults(
       }
       final AuthenticationResult authenticationResult = AuthorizationUtils.authenticationResultFromRequest(req);
 
-      if (offset != null && offset < 0) {
+      if (page == null) {
         return buildNonOkResponse(
             DruidException.forPersona(DruidException.Persona.USER)
                           .ofCategory(DruidException.Category.INVALID_INPUT)
                           .build(
-                              "offset cannot be negative. Please pass a positive number."
+                              "Page cannot be null."
                           )
         );
       }
-      if (numberOfRows != null && numberOfRows < 0) {
+      if (page < 0) {
         return buildNonOkResponse(
             DruidException.forPersona(DruidException.Persona.USER)
                           .ofCategory(DruidException.Category.INVALID_INPUT)
                           .build(
-                              "numRows cannot be negative. Please pass a positive number."
+                              "Page cannot be negative. Please pass a positive number."
                           )
         );
       }
 
```

Comment:
If the page does not exist its also an error. 
For eg: pageId=999 is passed but we only have 2 pages in the result, its also a invalid input.

Reply:
I have added this check below as "if page > 0" for the report since we don't really have a list of pages to check the index for yet and manually creating a list of a single page to check this doesn't make sense